includes/server/class-llms-rest-access-plans-controller.php

```diff
@@ -1028,4 +724,33 @@ private function related_product_permissions_check( $has_permissions, $request )
 		return $has_permissions;
 	}
 
+	/**
+	 * Block request when the access plan limit per product is reached.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error
+	 */
+	private function block_request_when_access_plan_limit_reached( $request ) {
+
+		$product_id = isset( $request['post_id'] ) ? $request['post_id'] : $this->get_object( (int) $request['id'] )->get( 'product_id' );
+		$product    = new LLMS_Product( $product_id );
+		$limit      = $product->get_access_plan_limit();
+
+		if ( count( $product->get_access_plans( false, false ) ) >= $limit ) {
+
+			return llms_rest_bad_request_error(
+				sprintf(
+					// Translators: %1$d = access plans limit per product, %2$s access plan post type plural name, %3$s product post type singular name.
+					__( 'Only %1$d %2$s allowed per %3$s', 'lifterlms' ),
+					$limit,
+					strtolower( get_post_type_object( $this->post_type )->labels->name ),
+					strtolower( get_post_type_object( get_post_type( $product_id ) )->labels->singular_name )
```

Comment:
there's a chance someone could filter this down to 1 access plan and create a language issue here:

"Only 1 access plans allowed per course"

however, the increase in code to accommodate this unlikely scenario seems silly to me. Right?

Let's just leave it

Reply:
Yeah I considered it, I wanted to use `_n()`, maybe I should note that I abandoned that idea because of the complexity.

Reply:
Yep, we're in agreement then!